### ISSUE TITLE
Pattern matching misleading error

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#5375]
```haxe
enum E {
	A;
	B( x : Int );
	C;
}

class Main {


	static function main() {
		var e : E = null;
		switch( e ) {
		case A:
		case B:
		case C:
		}
	}
	 
}
```

The following code outputs this:
```
Main.hx:14: characters 7-8 : Capture variables must be lower-case
Main.hx:15: characters 7-8 : Warning : This pattern is unused
```

I think this is a regression wrt 3.1 or 3.2, as I think it is supposed to complain about missing parameters for `B`

**Comment by Simn** [comment]
I acknowledge the issue but I'm reluctant to fix this for 3.3. It's not clear to me in which cases we should recover from an identifier that ends up being typed badly. For instance, we don't want to block off using a lower-case identifier if there's a local variable with the same name.

**Comment by Simn** [comment]
I went ahead and fixed this after all because it could lead to very weird behavior when lower-case identifiers are involved. I'm still not completely sure about the approach here, so let me know if there's something wrong still.
